pyproject.toml

```diff
@@ -27,55 +27,44 @@ classifiers = [
   "Programming Language :: Python :: 3.11",
   "Typing :: Typed",
 ]
-dynamic = ["version"]
-dependencies = [
-    "loguru >= 0.7.0",
-]
 
-[project.optional-dependencies]
-# Use `constraints.txt`
-tests = [
-  "coverage",
-  "pytest",
-  "pytest-cov",
-]
-docs = [
-  "furo",
-  "myst-parser",
-  "Sphinx",
-  "sphinxcontrib-spelling",
-]
-stubs = [
-  # Typing stubs go here
-]
-typing = [
-  "crustypy-instance[stubs]",
-  "mypy",
-  "mypy-extensions",
-  "nox",
-  "pytest",
-]
-dev = [
-  "crustypy-instance[tests,docs,stubs,typing]",
-  "black",
-  "build",
-  "nox",
-  "pre-commit",
-  "python-semantic-release",
-  "ruff",
-  "semver",
-  "toml",
-]
+[tool.poetry.dependencies]
+python = "^3.8"
+loguru = "^0.7.0"
+
+[tool.poetry.group.tests.dependencies]
+coverage = "^7.2.7"
+pytest = "^7.4.0"
+pytest-cov = "^4.1.0"
+
+[tool.poetry.group.docs.dependencies]
+furo = "^2023.5.20"
+myst-parser = "^2.0.0"
+Sphinx = "^7.0.1"
+sphinxcontrib-spelling = "^8.0.0"
+
+[tool.poetry.group.stubs.dependencies]
+# Typing stubs go here
+
+[tool.poetry.group.typing.dependencies]
+# Also requires group `stubs`
+mypy = "^1.4.0"
+mypy-extensions = "^1.0.0"
+nox = "^2023.4.22"
+pytest = "^7.4.0"
 
-[project.urls]
-"Homepage" = "https://github.com/MicaelJarniac/crustypy-instance"
-"Changelog" = "https://github.com/MicaelJarniac/crustypy-instance/blob/main/docs/CHANGELOG.md"
-"Documentation" = "https://crustypy-instance.readthedocs.io"
-"Repository" = "https://github.com/MicaelJarniac/crustypy-instance"
-"Bug Tracker" = "https://github.com/MicaelJarniac/crustypy-instance/issues"
+[tool.poetry.group.dev.dependencies]
+black = "^23.3.0"
+nox = "^2023.4.22"
+pre-commit = "^3.3.3"
+python-semantic-release = "^7.34.6"
+ruff = "^0.0.275"
+semver = "^3.0.1"
+toml = "^0.10.2"
```

Comment:
why would you need this package? it's old, unmaintained and iirc doesn't support the latest TOML standards

Reply:
That's a good question, IIRC one of the tools depends on it for some functionality, but I can't remember exactly which one. Will try to find it and see if they still require it.

Reply:
if it's not a top level dependency, no need to list it here, it will be resolved into `poetry.lock` anyway.

Reply:
IIRC the package that required it didn't list it as a dependency, it was more like "hey, if you want this tool to read `pyproject.toml`, you better make sure you installed `toml` yourself, good luck!".

Reply:
I figured it out I think, it was a Cruft dependency.
https://github.com/MicaelJarniac/crustypy/commit/409c90ada4aaf0e20feff117dace98a447363ff3

I don't even have Cruft in the dependencies anymore, so this was left over. Thanks for spotting it!